### PR TITLE
DEV-4343 | update changed property names in in new portal contributions list and detail view

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

Initially, this PR takes DEV-4032... as its base, as that is the PR/ticket that changes API schema/field names, with which this PR aligns the SPA.   DEV-4032 has not yet been merged to main. Once that has happened, base should change to main.

This is also to say that this PR is blocked for merge by DEV-4032.

#### What's this PR do?

In working on DEV-4032, I discovered some fields in portal contributions API endpoint schema that could be retired, and others where renaming made sense.  

This PR aligns the SPA with the API schema of backend, as described [here in this API schema file](https://github.com/newsrevenuehub/rev-engine/blob/f69b1d2a4c3272392916b9208fb632b2d2f62de9/apps/api/schemas/portal-contribution-list.openapi.yml).

Specifically:

- `credit_card_expiration_date` becomes `card_expiration_date` (to standardize all CC-related fields as being prefixed with `card_`)
- `last4` becomes `card_last_4`
- `id` is added as a new field
- `payment_provider_id` is removed, as it is not needed
- `provider_customer_id` is removed, as it is not needed

Additionally, this PR opportunistically solves for [DEV-4427](https://news-revenue-hub.atlassian.net/browse/DEV-4427)

#### Why are we doing this? How does it help us?

Part of shipping new portal.

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?

Yes.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

No, but existing ones have been updated to reflect new schema and updated type.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

Sort of. The API schema referenced in description above documents the schema of data returned by endpoint (though that happens in a different PR).

#### What are the relevant tickets? Add a link to any relevant ones.

- [DEV-4343](https://news-revenue-hub.atlassian.net/browse/DEV-4343)
- [DEV-4032](https://news-revenue-hub.atlassian.net/browse/DEV-4032)
- [DEV-4427](https://news-revenue-hub.atlassian.net/browse/DEV-4427)

#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No

[DEV-4343]: https://news-revenue-hub.atlassian.net/browse/DEV-4343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DEV-4427]: https://news-revenue-hub.atlassian.net/browse/DEV-4427?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by benjaminEwhite** [comment]
Closing and reopening because changed upstream migrations require rebuild.

**Review by nrh-cklimas — APPROVED** [review]
Code changes look good; the ContributionItem Storybook file needs an update to use the correct field for the last 4 digits of the card in the story data. See the Chromatic check on the PR for details. Once that's fixed, no need for re-review.

**Comment by benjaminEwhite** [comment]
Thanks @nrh-cklimas. I updated the storybook file.

**Review by x110dc — APPROVED** [review]

File: spa/src/components/portal/ContributionsList/ContributionsList.tsx [hunk 1]
```diff
@@ -13,9 +13,9 @@ export function ContributionsList() {
   const { contributionId } = useParams<{ contributionId?: string }>();
   const { contributor } = usePortalAuthContext();
   const { contributions, isError, isLoading, refetch } = usePortalContributionList(contributor?.id);
-  const selectedContribution = contributions.find(
-    (contribution) => contribution.payment_provider_id === contributionId
-  );
+  const selectedContribution = contributionId
```
Comment (nrh-cklimas) [inline spa/src/components/portal/ContributionsList/ContributionsList.tsx:16]: If `find` doesn't match anything, it returns `undefined`. Any reason to coerce this to `null`?

**Review by nrh-cklimas — APPROVED** [review]

**Pull request merged**